pymllm/layers/attention/gdn_chunkwise.py

```
@@ -0,0 +1,429 @@
+    V: int,
+    device: torch.device,
+) -> torch.Tensor:
+    """Process one chunk: WY construction → matmul output → state update."""
+
+    # ---- cumulative decay within chunk ----
+    log_alpha_c = torch.log(alpha_c.clamp(min=1e-10))  # [C, HV]
+    cum_log = torch.cumsum(log_alpha_c, dim=0)          # [C, HV]
+    gamma = torch.exp(cum_log)                           # [C, HV]
+    gamma_C = gamma[-1]                                  # [HV]
+
+    # ---- precompute β·k and (β/γ)·v ----
```

Ruff (0.15.15)

[warning] Comment contains ambiguous `γ` (GREEK SMALL LETTER GAMMA). Did you mean `y` (LATIN SMALL LETTER Y)?

(RUF003)